### PR TITLE
feat(cli): check version against core

### LINKED ISSUE #390: fix: assert CLI version matches core
We should error or at least warn if the CLI version used does not match the installed core version.

### PR DESCRIPTION
**Pull request by AndersDJohnson**
Adds to CLI boot routine for all commands to check CLI version against user-installed mockyeah core version to ensure no mismatch. In future we could use semver compare, but for now since we still in unstable `0.x`, we'll require exact match.

Also introduces `jest` as a unit test framework for the CLI - core uses `mocha` now but we could eventually migrate to `jest` for both.

Fixes #390.

### CODE REVIEW

File: packages/mockyeah-cli/lib/__tests__/boot.test.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          package: {
+            version: '0.0.0-test'
+          }
+        };
+        checkVersionMatchWithPackage(env, pkgUp);
+      }).toThrow(
+        'Version mismatch between CLI (0.0.0-test) and core (99.99.99-test) - please install same versions.'
+      );
+    });
+  });
```
Comment (ryanricard) [inline packages/mockyeah-cli/lib/__tests__/boot.test.js]: No positive test case?

File: packages/mockyeah-cli/.eslintrc.js [hunk 2]
```diff
@@ -1 +1,7 @@
-module.exports = require('mockyeah-tools/.eslintrc.js');
+const tools = require('mockyeah-tools/.eslintrc.js');
+
+module.exports = Object.assign({}, tools, {
+  env: Object.assign({}, tools.env, {
+    jest: true
+  })
+});
```
Comment (ryanricard) [inline packages/mockyeah-cli/.eslintrc.js]: Nice!

File: packages/mockyeah-cli/lib/boot.js [hunk 3]
```diff
@@ -33,4 +67,8 @@ module.exports = function boot(callback) {
 
     callback.call(this, env);
   });
-};
+}
+
+boot.checkVersionMatchWithPackage = checkVersionMatchWithPackage;
```
Comment (ryanricard) [inline packages/mockyeah-cli/lib/boot.js]: Should we externalize `checkVersionMatchWithPackage` into lib? I get a bad smell when methods are exported just for testing.

File: packages/mockyeah-cli/lib/__tests__/boot.test.js (hunk 1 above)
Comment (AndersDJohnson) [reply packages/mockyeah-cli/lib/__tests__/boot.test.js]: @ryanricard Excellent callout. We should look at wiring up code coverage here too somehow sometime. Added a positive test case in dc1ad34.

File: packages/mockyeah-cli/lib/boot.js (hunk 3 above)
Comment (AndersDJohnson) [reply packages/mockyeah-cli/lib/boot.js]: @ryanricard Probably. Can refactor. Hold up.

Comment (AndersDJohnson) [reply packages/mockyeah-cli/lib/boot.js]: @ryanricard OK, refactored to external file in baaa05a.

File: packages/mockyeah-cli/lib/checkVersionMatch.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  if (cliVersion !== coreVersion) {
+    throw new Error(
+      chalk.red(
+        `Version mismatch between CLI (${cliVersion}) and core (${coreVersion}) - please install same versions.`
+      )
+    );
+  }
+};
+
+module.exports = checkVersionMatch;
```
Comment (ryanricard) [inline packages/mockyeah-cli/lib/checkVersionMatch.js]: Much better :)

**Pull request merged**